frontend/src/main/web/lib/components/UserProfile.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import RecentContributions from './RecentContributions';
+import Configs from '../constants/Configs';
+import UserMatrixStore from '../stores/UserMatrixStore';
+
+var UserProfile = React.createClass({
+
+  getMatrixState: function() {
+    return UserMatrixStore.getMatrixState();
+  },
+
+  getInitialState: function() {
+    return this.getMatrixState();
+  },
+
+  componentDidMount: function() {
+    UserMatrixStore.addChangeListener(this._onChange);
+  },
+
+  componentWillUnmount: function() {
+    UserMatrixStore.removeChangeListener(this._onChange);
+  },
+
+  _onChange: function() {
+    this.setState(this.getMatrixState());
+  },
+
+  render: function() {
+    var user = Configs.data.profileUser,
+      authenticated = Configs.user ? Configs.user.loggedIn : false,
+      recentContribution = (<div></div>),
+      imageUrl = '',
+      name = '',
+      languageTeams = '',
+      username;
+
+    if(authenticated) {
+      recentContribution =
+        (<div className="g__item w--3-4 w--1-2-m">
+          <div className="bg--pop-highest l--pad-v-1">
+            <RecentContributions
+              loading={this.state.loading}
+              dateRangeOption={this.state.dateRangeOption}
+              matrixForAllDays={this.state.matrixForAllDays}
+              wordCountsForSelectedDayFilteredByContentState={this.state.wordCountsForSelectedDayFilteredByContentState}
+              wordCountsForEachDayFilteredByContentState={this.state.wordCountsForEachDayFilteredByContentState}
+              contentStateOption={this.state.contentStateOption}
+              selectedDay={this.state.selectedDay}
+              dateRange={this.state.dateRange}/>
```

Comment:
You should be able to replace all these attributes with spread operator:

`<RecentContributions {...this.state}/>`

The only concern is if there are other properties in `this.state` that you do not want to send to the component - in that case I recommend `lodash.pick` or `lodash.omit` to concisely get just the props you want to a new object, then use `{...` with it.